artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/ParameterisedAddress.java

```diff
@@ -106,4 +107,46 @@ public static boolean isParameterised(SimpleString address) {
       return URISupport.containsQuery(address);
    }
 
+   public static SimpleString extractAddress(SimpleString address) {
+      return SimpleString.toSimpleString(extractAddress(address.toString()));
+   }
+
+   /**
+    * Given an address string, extract only the query portion if the address is
+    * parameterized, otherwise return an empty {@link Map}.
+    *
+    * @param address
+    *       The address to operate on.
+    *
+    * @return a {@link Map} containing the parameters associated with the given address.
+    */
+   @SuppressWarnings("unchecked")
+   public static Map<String, String> extractParameters(String address) {
+      final int index = address.indexOf('?');
+
+      if (index == -1) {
+         return Collections.EMPTY_MAP;
+      } else {
+         return parseQuery(address);
+      }
+   }
+
+   /**
+    * Given an address string, extract only the address portion if the address is
+    * parameterized, otherwise just return the provided address.
+    *
+    * @param address
+    *       The address to operate on.
+    *
+    * @return the original address minus any appended parameters.
+    */
+   public static String extractAddress(String address) {
+      final int index = address.indexOf('?');
+
+      if (index == -1) {
+         return address;
+      } else {
+         return address.substring(0, index);
+      }
+   }
```

Comment:
The methods cant handle a null but dont say that and the 'otherwise return provided' bit might imply otherwise. Should either say so or handle it.